### PR TITLE
[[ Bug 19032 ]] Uncache object IDs when removing from object tree

### PR DESCRIPTION
**Pull request by livecodeali**

### CODE REVIEW

File: engine/src/object.cpp [hunk 1]
```diff
@@ -317,11 +317,6 @@ MCObject::~MCObject()
 	MCMemoryDeleteArray(patterns);
 	deletepropsets();
 	MCValueRelease(tooltip);
-
-	// MW-2012-11-20: [[ IdCache ]] Make sure we delete the object from the cache - not
-	//   all deletions vector through 'scheduledelete'.
-	if (m_in_id_cache)
```
Comment (runrevmark) [inline engine/src/object.cpp:323]: I suggest putting MCAssert(!m_in_id_cache) here to ensure there aren't any odd cases :)

**Comment by runrevmark** [comment]
@livecode-vulcan review okay 1926bd4

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 1926bd4fb3591138d9cb84f8f6536a209bf25702

**Comment by livecode-vulcan** [comment]
😎 test success 1926bd4fb3591138d9cb84f8f6536a209bf25702

* try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/817)
* try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/822)
* try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/815)
* try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/816)
* try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/822)
* try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/814)
* try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/814)

**Pull request merged**